### PR TITLE
Enums comparator with ignore case

### LINKED ISSUE #8084: [REQ] Feature Request Description
#### Java
Currently, it is not possible to customize enum case sensitiveness in the generated code.
It was noted on the [StackOverflow](https://stackoverflow.com/questions/62130956/openapi-generator-case-insensitive-configuration-for-enum-deserialization) also.

There is a generated method, currently, that is used for serialization / deserialization:

```java
public static ResourceType fromValue(String value) {
    for (ResourceType b : ResourceType.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }
```

It causes problems while serialization processes

## Describe the solution you'd like

1. equalsIngoreCase for `fromValue` always

```java
public static ResourceType fromValue(String value) {
    for (ResourceType b : ResourceType.values()) {
      if (b.value.equalsIgnoreCase(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }
```

2. Have a property to customize it

`caseInsensitiveEnum: true` //default is false to keep BC

## Describe alternatives you've considered

* Always and only use upper case on client and server side

### PR DESCRIPTION
**Pull request by wtrocki**
fixes #8084 #8084

Enum comparison in Java should use ignore case to handle values passed using capitals (More info in linked issues)

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wtrocki** [comment]
Working on issue on my end with generating samples.

**Comment by wtrocki** [comment]
Looks like python examples on main require specific python version and have been failing on my end

File: modules/openapi-generator/src/main/resources/Java/modelEnum.mustache [hunk 1]
```diff
@@ -56,7 +56,7 @@ import com.google.gson.stream.JsonWriter;
 {{/jackson}}
   public static {{{datatypeWithEnum}}}{{^datatypeWithEnum}}{{{classname}}}{{/datatypeWithEnum}} fromValue({{{dataType}}} value) {
     for ({{{datatypeWithEnum}}}{{^datatypeWithEnum}}{{{classname}}}{{/datatypeWithEnum}} b : {{{datatypeWithEnum}}}{{^datatypeWithEnum}}{{{classname}}}{{/datatypeWithEnum}}.values()) {
-      if (b.value.equals(value)) {
+      if (b.value.{{^isString}}equals{{/isString}}{{#isString}}equalsIgnoreCase{{/isString}}(value)) {
```
Comment (borsch) [inline modules/openapi-generator/src/main/resources/Java/modelEnum.mustache:59]: as for me you should compare enum's constant name. so should be something like this

`if (b.value.equals(value){{#isString}} || b.name().equals(value){{/isString}}) {`

**Comment by rodrigoma3** [comment]
@wtrocki I'm in need of this fix, could you finalize and merge? Or authorize me to continue from here on your branch?

**Comment by wtrocki** [comment]
@wing328 Would you be willing to accept that PR - happy to rebase.

**Comment by rodrigoma3** [comment]
@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)
Can we go ahead with this fix?

**Comment by wtrocki** [comment]
@rodrigoma3 feel free to open new PR based off my changes

**Comment by borsch** [comment]
@wtrocki can you rebase this PR?

File: samples/client/echo_api/java/apache-httpclient/src/main/java/org/openapitools/client/ApiClient.java [hunk 2]
```diff
@@ -77,7 +77,7 @@
 
 import org.openapitools.client.auth.Authentication;
 
-@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaClientCodegen")
+@.annotation.Generated(value = "org.openapitools.codegen.languages.JavaClientCodegen")
```
Comment (borsch) [inline samples/client/echo_api/java/apache-httpclient/src/main/java/org/openapitools/client/ApiClient.java:80]: this is weird

File: samples/client/echo_api/java/apache-httpclient/.openapi-generator/VERSION [hunk 3]
```diff
@@ -1 +1 @@
-6.6.0-SNAPSHOT
\ No newline at end of file
+6.3.0-SNAPSHOT
```
Comment (borsch) [inline samples/client/echo_api/java/apache-httpclient/.openapi-generator/VERSION:1]: pull latest master

**Review by borsch — CHANGES_REQUESTED** [review]

File: samples/client/echo_api/java/apache-httpclient/.openapi-generator/VERSION (hunk 3 above)
Comment (wtrocki) [reply samples/client/echo_api/java/apache-httpclient/.openapi-generator/VERSION:1]: I though I did. Will do now.

**Comment by borsch** [comment]
closing since since similar changes were merged in https://github.com/OpenAPITools/openapi-generator/pull/15273

**Pull request closed without merging**